### PR TITLE
Add onboarding view and sign in with Apple button

### PR DESCRIPTION
**Pull request by juanlaube**
This adds an onboarding view with two buttons to sign in with Apple and login with email.
The sign in with Apple button is a View that wraps the WKInterface object for the actual sign in with Apple button and has a callback to handle the token or an error, the other button doesn't do anything for now.
I will add state, actions and reducers in a different pull request.

### CODE REVIEW

**Comment by juanlaube** [comment]
@sofokli1 fixed! There's one detail though, when you keep pressing on the  button you can notice some clipping on the left and right, I've spent some time trying to fix it but couldn't find a solution.

**Comment by sofokli1** [comment]
personally I am ok with making the email button have same corner radius as the apple one to fix the clipping. @Odrakir what do you think?

**Comment by Odrakir** [comment]
- The buttons are cut (screenshot attached)
- Sometimes tapping the Sign In with Apple button take me to the regular log in screen... 🤷‍♂ 
- We need a clear message when this fails due to user not being already signed up

IMG_5056

Also, I keep getting an error

File: TogglTrack/Core/Actions/LoginAction.swift [hunk 1]
```diff
@@ -11,6 +11,7 @@ import Foundation
 public enum LoginAction
 {
     case login(String, String)
+    case startLoginWithApple
     case loginWithApple(String)
```
[the remark this answers is not in the document]
Comment (juanlaube) [reply TogglTrack/Core/Actions/LoginAction.swift]: I renamed `startLoginWithApple` to `getLoginWithAppleToken`.

File: TogglTrack/Core/Reducers/LoginReducer.swift [hunk 2]
```diff
@@ -98,6 +106,14 @@ private func loginEffect(_ api: APIProtocol, _ token: String) -> Effect<AppActio
         )
 }
 
+private func loginWithAppleEffect(_ signInWithApple: SignInWithAppleServiceProtocol) -> Effect<AppAction>
```
[the remark this answers is not in the document]
Comment (juanlaube) [reply TogglTrack/Core/Reducers/LoginReducer.swift]: I renamed it to `getLoginWithAppleTokenEffect`.

**Comment by sofokli1** [comment]
@juanlaube is there a consensus regarding the copy? `Sign In` vs `Log in`

**Comment by juanlaube** [comment]
Not really, but since Apple uses "sign in" everywhere, it shows in the UI and we cannot change it, I went with that.

**Comment by sofokli1** [comment]
Can you also change email login to say `sign in` or you feel like that is out side of the scope?

**Comment by juanlaube** [comment]
Oops, I forgot to push something 🙊 
Check again @sofokli1

**Pull request merged**